File: _Src/Freem/Freem.Web/src/clients/models/users/LoginPassword/RegisterPasswordCredentialsRequest.ts
export class RegisterPasswordCredentialsRequest {
  public readonly nickname: string;
  public readonly login: string;
  public readonly password: string;

  constructor(nickname: string, login: string, password: string) {
    this.nickname = nickname;
    this.login = login;
    this.password = password;
  }
}

File: _Src/Freem/Freem.Web.Api.Public.Contracts/DTO/Users/LoginPassword/RegisterPasswordCredentialsRequest.cs
using Freem.Entities.Users.Models;

namespace Freem.Web.Api.Public.Contracts.DTO.Users.LoginPassword;

public class RegisterPasswordCredentialsRequest
{
    public Nickname Nickname { get; }
    public Login Login { get; }
    public Password Password { get; }

    public RegisterPasswordCredentialsRequest(Nickname nickname, Login login, Password password)
    {
        ArgumentNullException.ThrowIfNull(nickname);
        ArgumentNullException.ThrowIfNull(login);
        ArgumentNullException.ThrowIfNull(password);
        
        Nickname = nickname;
        Login = login;
        Password = password;
    }
}